DB/common.php

```diff
@@ -1148,7 +1148,14 @@ function freePrepared($stmt, $free_resource = true)
      */
     function modifyQuery($query)
     {
-        return $query;
+
+      if (class_exists('Civi\Core\Container') && \Civi\Core\Container::isContainerBooted()) {
+        Civi\Core\Container::singleton()->get('dispatcher')->dispatch('civi.db.query',
+          \Civi\Core\Event\GenericHookEvent::create(array('query' => &$query))
```

Comment:
@eileenmcnaughton Regarding performance, the `dispatch()` should be pretty cheap on non-hook events.

Since this isn't actually a hook, and since we're likely to fire O(100)s of times, I'd lean pretty solidly against using `GenericHookEvent` for this. Instead, make a class that doesn't use any reflection or array-mangling, like:

```php
namespace Civi\Core\Event;

/**
 * Class QueryEvent
 * @package Civi\Event
 */
class QueryEvent extends \Symfony\Component\EventDispatcher\Event {

  /**
   * @var string
   */
  public $query;

  /**
   * QueryEvent constructor.
   * @param string $query
   */
  public function __construct($query) {
    $this->query = $query;
  }

}
```

Reply:
@totten lol - that was code you proposed in an earlier conversation - but OK - I think I see what you say we should change it to - let me try